smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java

```diff
@@ -106,11 +111,28 @@ void init() {
         this.vertx = null;
     }
 
-    private synchronized AmqpClient getClient(Config config) {
-        // TODO Should we support having a single client (1 host) or multiple clients.
-        if (client != null) {
-            return client;
+    private AmqpClient createClient(Config config) {
+        AmqpClient client;
+        Optional<String> clientOptionsName = config.getOptionalValue("client-options-name", String.class);
+        if (clientOptionsName.isPresent()) {
+            String optionsName = clientOptionsName.get();
+            Instance<AmqpClientOptions> options = clientOptions.select(NamedLiteral.of(optionsName));
+            if (options.isUnsatisfied()) {
+                throw new IllegalStateException(
+                        "Cannot find a " + AmqpClientOptions.class.getName() + " bean named " + optionsName);
+            }
+            LOGGER.debug("Creating amqp client from bean named " + optionsName);
+            client = AmqpClient.create(new io.vertx.axle.core.Vertx(vertx.getDelegate()), options.get());
+        } else {
+            //TODO pick a default client options if there is one and use it, using a specific qualifier? like @DefaultAmqpConfiguration ?
+            //backwards compatibility
```

Comment:
Good question, I would fall back to the current behavior, meaning ignore the options.
Maybe issue a warning.

Reply:
do you mean to create an issue to discuss it?

Reply:
Yes please do.

Reply:
done, I also removed those comments so this PR could be ready to be merged